core/log-core.js

```diff
@@ -774,6 +782,57 @@ var isValidObject = function (obj, canBeEmpty) {
     return true;
 };
 
+// check if the given object is an Error with stacktrace using duck typing
+var isErrorWithStacktrace = function (obj) {
+    if (obj && obj.stack && obj.message && typeof obj.stack === "string" && typeof obj.message === "string") {
+        return true;
+    }
+    return false;
+}
+
+// Split stacktrace into string array and truncate lines if required by size limitation
+// Truncation strategy: Take one line from the top and two lines from the bottom of the stacktrace until limit is reached.
+var prepareStacktrace = function (stacktraceStr) {
+    var fullStacktrace = stacktraceStr.split('\n');
+    var totalLineLength = fullStacktrace.reduce((acc, line) => acc + line.length, 0);
+
+    if (totalLineLength > MAX_STACKTRACE_SIZE) {
+        var truncatedStacktrace = [];
+        var stackA = [];
+        var stackB = [];
+        var indexA = 0;
+        var indexB = fullStacktrace.length - 1;
+        var currentLength = 73; // set to approx. character count for "truncated" and "omitted" labels
+
+        for (var i = 0; i < fullStacktrace.length; i++) {
+            if (i % 3 == 0) {
+                var line = fullStacktrace[indexA];
```

Comment:
> why is every variable defined before, and only line inside the loop?

Typically I prefer to have variable definitions close to where I use them. `line` could be defined outside the loop, but there is no real benefit in doing so. None of the two branches within the loop accesses its value before writing it, so values from previous iteration do not need to be accessible in subsequent iterations.
However, I think it would make sense to use `let` instead of `var` here to make `line`block scoped for both branches.